homework/tests/example.spec.js

```diff
@@ -15,17 +15,712 @@ const data = [
+            await page.selectOption('#selectOperationDropdown', {label: 'Divide'});
+            await page.locator('#integerSelect').check();
+            await page.locator('#calculateButton').click();
+      
+            await expect(page.locator('#numberAnswerField')).toHaveValue('0');
+        });
+    
+        test('Should show error message when a letter is typed in first number field: a and 3', async ({page}) => {
+          await page.goto('https://testsheepnz.github.io/BasicCalculator');
+            await page.selectOption('#selectBuild', { label: version});
+            await page.locator('#number1Field').type('a');
+            await page.locator('#number2Field').type('3');
+            await page.selectOption('#selectOperationDropdown', {label: 'Divide'});
+            await page.locator('#calculateButton').click();
+      
+            await expect(page.locator('#errorMsgField')).toHaveText('Number 1 is not a number');
+        });
+      
+        test('Should show error message when a letter is typed in second number field: 2 and a', async ({page}) => {
+          await page.goto('https://testsheepnz.github.io/BasicCalculator');
+            await page.selectOption('#selectBuild', { label: version});
+            await page.locator('#number1Field').type('2');
+            await page.locator('#number2Field').type('a');
+            await page.selectOption('#selectOperationDropdown', {label: 'Divide'});
+            await page.locator('#calculateButton').click();
+      
+            await expect(page.locator('#errorMsgField')).toHaveText('Number 2 is not a number');
+        });
+    
+    
+    
+        
+    
+      });
+    
+      test.describe.parallel(version + ': Concatenation', () => {
+        test('Should concatenate two letters: a and b', async ({page}) => {
+          await page.goto('https://testsheepnz.github.io/BasicCalculator');
+            await page.selectOption('#selectBuild', { label: version});
+            await page.locator('#number1Field').type('a');
+            await page.locator('#number2Field').type('b');
+            await page.selectOption('#selectOperationDropdown', {label: 'Concatenate'});
+            await page.locator('#calculateButton').click();
+      
+            await expect(page.locator('#numberAnswerField')).toHaveValue('ab');
+        });
+    
+        test('Should concatenate two numbers: 2 and 3', async ({page}) => {
+          await page.goto('https://testsheepnz.github.io/BasicCalculator');
+            await page.selectOption('#selectBuild', { label: version});
+            await page.locator('#number1Field').type('2');
+            await page.locator('#number2Field').type('3');
+            await page.selectOption('#selectOperationDropdown', {label: 'Concatenate'});
+            await page.locator('#calculateButton').click();
+      
+            await expect(page.locator('#numberAnswerField')).toHaveValue('23');
+        });
+    
```

Comment:
This timeout message "Test timeout of 10000ms exceeded." does not really provide a lot of information for us. When the test fail like this - we should check what is an issue, register it and improve the test so next time we will know that "Calculate" button is missing just looking into the report.

This could be achieved by using if() & else() blocks, for example:

if (await page.locator('#number1Field').isVisible()) { await page.locator('#number1Field').type('2'); } else{ throw "First number input field not working/missing"; }
Now if you would use that, message "First number input field not working/missing" would be displayed in the report, instead of timeout message. However, this is not a good practice to use those checks in every single test, so please update your code:

Try to exclude those checks into the separate function and call it in every test, if those checks fails - further test code will not be executed
If you will not be able to come up with solution described in 1st point, update only your last test "Concatenating 1 and 2 results in 12" with those checks.

Your report is not visible to me, so please double-check if it works as expected. Attaching a screenshot from fellow student Laurynas, I am sure you will have exact same issue.

![image](https://user-images.githubusercontent.com/73665598/180805521-2defe5f3-7e54-4046-91c6-c35c5d982099.png)